src/hotspot/share/gc/z/zVerify.cpp

```diff
@@ -373,14 +373,12 @@ class ZVerifyBadOopClosure : public OopClosure {
 // frame iteration code, such as creating random handles even though there
 // is no safepoint to protect against, and fiddling around with exceptions.
 class StackWatermarkProcessingMark {
-  ResetNoHandleMark     _rnhm;
```

Comment:
Since you're removing the ResetNoHandleMark, you should revisit
the comment block for the class. The entire sentence to revisit is:
```
// ... It is mostly
// due to problems that we might want to eventually clean up inside of the
// frame iteration code, such as creating random handles even though there
// is no safepoint to protect against, and fiddling around with exceptions.
```

Reply:
This statement is still true.  PreserveExceptionMark creates a Handle, so we need the HandleMark.  We don't need to ResetNoHandleMark with this change.

PreserveExceptionMark::PreserveExceptionMark(Thread*& thread) {
  thread     = Thread::current();
  _thread    = thread;
  _preserved_exception_oop = Handle(thread, _thread->pending_exception());

(which will create a Handle if there's a pending_exception).